src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -21,14 +23,17 @@ public class IexService {
   @NonNull
   private IexClient iexClient;
 
+  @Value("${spring.rest.token}")
+  private String token;
```

Comment:
This property should live in the IexClient class -- I'd consider it a query-layer detail that the client should abstract away from the service. The "hint" that it should live one layer down is the fact you had to change every method call in this class to pass the same thing in.

Reply:
I've moved the token property down to the IexClient interface. It works, but I don't think there's a way to use the @Value annotation in an interface, which I'm currently using to retrieve the token value from application.yml. The API token is hardcoded now in the interface, but there might be a better way that I'm not aware of.  thoughts on this solution?

Reply:
Not a fan because of the hard-coding. I think the best way to actually achieve what we want here is to configure a RequestInterceptor for the feign client, and then for each request add the token query parameter. 

https://www.baeldung.com/spring-cloud-openfeign#bd-1-implementing-requestinterceptor

Reply:
Adding a request interceptor to add the token worked, thanks!